vm/src/dictdatatype.rs

```diff
@@ -191,27 +197,40 @@ impl<T: Clone> Dict<T> {
     }
 
     pub fn clear(&self) {
-        let mut inner = self.borrow_value_mut();
-        inner.entries.clear();
-        inner.indices.clear();
-        inner.size = 0
+        let mut swapped = Vec::new();
+        {
+            let mut inner = self.borrow_value_mut();
+            // defer dec rc
+            std::mem::swap(&mut inner.entries, &mut swapped);
```

Comment:
I don't think you need the inner block, since it's at the end of the function anyway. Also, I think you could just do something like `let _old = std::mem::replace(entries, Vec::new());` to defer til the end of scope.